docs/dev/contributing.md

```diff
@@ -89,11 +89,27 @@ If your commit is only shipping documentation changes or example files, and is a
 complete no-op for the test suite, please add **[skip ci]** in the commit message
 body to skip the build and give that slot to someone else who does need it.
 
-### Squash your commits
+### Pull request workflow
 
-Please rebase your changes on `main` and squash your commits whenever possible,
-it keeps history cleaner and it's easier to revert things. It also makes developers
-happier!
+The goals ordered by priority are:
+- Make PR reviews (both initial and follow-up reviews) easy for reviewers using GitHub
+- On the `main` branch, have a meaningful commit history that allows understanding (even years later) what each commit does, and why. 
+
+#### Before the first in-depth PR review
```

Comment:
Should we explicitly recommend to mark a PR as draft before it's ready for a review?

Reply:
I don't think so. In real life you simply don't open the PR until you're ready to receive reviews.

Only long running PR, where you want early feedback, get into this situation.

Reply:
I don't think there's disagreement here actually :)

I think we can add a recommendation that PRs should only be opened once they're reviewable, that _draft_ PRs should be used to share code before it's ready for actual reviews.